plugins/languages/rust/rustaceanvim/settings-options.nix

```
@@ -0,0 +1,294 @@
+          Whether the window gets automatically focused.
+        '';
+
+        open_split =
+          helpers.defaultNullOpts.mkEnumFirstDefault
+            [
+              "horizontal"
+              "vertical"
+            ]
+            ''
+              Whether splits opened from floating preview are vertical.
+            '';
+      };
+
+      crate_graph = {
+        backend = helpers.defaultNullOpts.mkStr "x11" ''
+          Backend used for displaying the graph.
+          See: https://graphviz.org/docs/outputs
+        '';
+
+        output = helpers.mkNullOrStr ''
+          Where to store the output.
+          No output if unset.
+          Relative path from `cwd`.
+        '';
+
+        full = helpers.defaultNullOpts.mkBool true ''
+          `true` for all crates.io and external crates, false only the local crates.
+        '';
+
+        enabled_graphviz_backends =
+          helpers.defaultNullOpts.mkListOf types.str
+            ''
+              [
+                "bmp" "cgimage" "canon" "dot" "gv" "xdot" "xdot1.2" "xdot1.4" "eps" "exr" "fig" "gd"
+                "gd2" "gif" "gtk" "ico" "cmap" "ismap" "imap" "cmapx" "imap_np" "cmapx_np" "jpg"
+                "jpeg" "jpe" "jp2" "json" "json0" "dot_json" "xdot_json" "pdf" "pic" "pct" "pict"
+                "plain" "plain-ext" "png" "pov" "ps" "ps2" "psd" "sgi" "svg" "svgz" "tga" "tiff"
+                "tif" "tk" "vml" "vmlz" "wbmp" "webp" "xlib" "x11"
+              ]
+            ''
+            ''
+              Override the enabled graphviz backends list, used for input validation and autocompletion.
+            '';
+
+        pipe = helpers.mkNullOrStr ''
+          Override the pipe symbol in the shell command.
+          Useful if using a shell that is not supported by this plugin.
+        '';
+      };
+
+      open_url = helpers.defaultNullOpts.mkLuaFn "require('rustaceanvim.os').open_url" ''
+        If set, overrides how to open URLs.
+        `fun(url:string):nil`
+      '';
+    };
+
+  server = {
+    auto_attach = helpers.mkNullOrStrLuaFnOr types.bool ''
+      Whether to automatically attach the LSP client.
+      Defaults to `true` if the `rust-analyzer` executable is found.
+
+      This can also be the definition of a function (`fun():boolean`).
+
+      Default:
+      ```lua
+        function(bufnr)
+          if #vim.bo[bufnr].buftype > 0 then
+            return false
+          end
+          local path = vim.api.nvim_buf_get_name(bufnr)
+          if not os.is_valid_file_path(path) then
+            return false
+          end
+          local cmd = types.evaluate(RustaceanConfig.server.cmd)
+          ---@cast cmd string[]
+          local rs_bin = cmd[1]
+          return vim.fn.executable(rs_bin) == 1
+        end
+      ```
+    '';
+
+    on_attach = helpers.mkNullOrLuaFn ''
+      Function to call on attach.
+      By default, it will use the Nixvim global `__lspOnAttach` function.
```

This should probably be clarified; maybe something along the lines of this?

```suggestion
      If `plugins.lsp` is enabled, it defaults to the Nixvim global `__lspOnAttach` function. Otherwise it defaults to `null`.
```

---

Agreed, done.